### DISCUSSION TITLE
K6 script execution shows errors like -> Error: No request with name 'requestName' and index 4

### DISCUSSION THREAD

**Discussion by prabodhkelkar** [apideck-libraries/postman-to-k6#125] [Q&A] [answered]
Hello,
In my postman collection, I do have couple requests repeating in different folders. So, after conversion to K6 using "postman-to-k6" utility (with --separate option), the K6 script generated has "group" section created for each folder and each repeating request gets some index number associated with it.

e.g. postman[Request]("getAllNotifications", 4);
--> my understanding here is --> above line indicates "getAllNotifications" request is getting executed fourth time

![image](https://github.com/apideck-libraries/postman-to-k6/assets/60377254/9ac01747-0504-476b-a97e-b565ea42f301)

All the requests reside in "requests" folder under and are referred in K6 scripts.
By mistake; I deleted one request in "requests" folder and added it back manually (created js file manually).

But, now when I am running the K6 script, I get error as below. I am not sure what I need to change to get rid of this error message.
Any clue?

![image](https://github.com/apideck-libraries/postman-to-k6/assets/60377254/2b5be8f7-c0ff-4c63-88d9-744720d28404)

![image](https://github.com/apideck-libraries/postman-to-k6/assets/60377254/b75bedd5-1589-4e5a-86dc-719e18aba9bf)

**Comment by thim81** [comment]
hi @prabodhkelkar 

The easiest way is to just rerun the postman-to-k6 with the --separate option.

**Reply by prabodhkelkar** [reply, marked as answer]
Thanks for the reply. But, I cannot rerun the postman-to-k6 as I did loads of changes to requests to make them work. Being complex graphQL queries with cookie based authentication, this required loads of changes - from switching to http.cookieJar to separate out queries for each request and JSON.stringify them as well as replace as date assignments that were based on moment().

But, I found the issue and fixed it. 
When I deleted the requests, I removed the import statements. Now, after adding back the requests (same request in 4 different folders, so request gets called from 4 different groups), I added back import statements. But missed to add import statement for request from second group/folder. While running the script, it did not show any issue for request in group 2, but because there were only 3 imports for request, it started showing error for 4th call of request. Details below will explain it better...

folder structure like - 
folder-1
- requestABC.js
folder-2
- requestABC.js
folder-3
- requestABC.js
folder-4
- requestABC.js

import statements should be like - 
import "./requests/folder-1/requestABC.js";
import "./requests/folder-2/requestABC.js"; // I missed to add this
import "./requests/folder-3/requestABC.js";
import "./requests/folder-4/requestABC.js";

Call for the requests is like - 
group("folder-1", function() {
      postman[Request]("requestABC"); }
group("folder-2", function() {
      postman[Request]("requestABC", 2); } // this started calling request from folder-3
group("folder-3", function() {
      postman[Request]("requestABC", 3); } // this started calling request from folder-4
group("folder-4", function() {
      postman[Request]("requestABC", 4); } // Started showing error as "No request with name requestABC and index 4"


Solution:
Adding the missing import statement for requestABC from folder-2 solved the issue.
-> import "./requests/folder-2/requestABC.js";

**Reply by thim81** [reply]
Thanks for sharing your updates and resolutions.
